partials/KafkaPublisherImpl.java

```diff
@@ -1,69 +1,69 @@
-        {%- endif %}
-    {% if channel.description() or channel.subscribe().description() %}/**{% for line in channel.description() | splitByLines %}
-     * {{line | safe}}{% endfor %}{% for line in channel.subscribe().description() | splitByLines %}
-     * {{line | safe}}{% endfor %}
-     */{% endif %}
-    public void {{methodName}}(Integer key, {{varName | upperFirst}} {{varName}}{% if hasParameters %}{%for parameterName, parameter in channel.parameters() %}, {% if parameter.schema().type() === 'object'%}{{parameterName | camelCase | upperFirst}}{% else %}{{parameter.schema().type() | toJavaType(false)}}{% endif %} {{parameterName | camelCase}}{% endfor %}{% endif %}) {
-        Message<{{varName | upperFirst}}> message = MessageBuilder.withPayload({{varName}})
-                .setHeader(KafkaHeaders.TOPIC, get{{methodName | upperFirst-}}Topic({% if hasParameters %}{%for parameterName, parameter in channel.parameters() %}{{parameterName | camelCase}}{% if not loop.last %}, {% endif %}{% endfor %}{% endif %}))
-                .setHeader(KafkaHeaders.{%- if params.springBoot2 %}MESSAGE_KEY{% else %}KEY{% endif -%}, key)
+
+    // Example method for publishing messages
+    public void publishMessage(Integer key, YourMessageType messagePayload) {
+        Message<YourMessageType> message = MessageBuilder.withPayload(messagePayload)
+                .setHeader(KafkaHeaders.TOPIC, "yourTopicName")
+                .setHeader(KafkaHeaders.KEY, key)
                 .build();
-        kafkaTemplate.send(message);
-    }
 
-    private String get{{methodName | upperFirst-}}Topic({% if hasParameters %}{%for parameterName, parameter in channel.parameters() %}{% if parameter.schema().type() === 'object'%}{{parameterName | camelCase | upperFirst}}{% else %}{{parameter.schema().type() | toJavaType(false)}}{% endif %} {{parameterName | camelCase}}{% if not loop.last %}, {% endif %}{% endfor %}{% endif %}) {
-        Map<String, String> parameters = {% if hasParameters %}new HashMap<>(){% else %}null{% endif %};
-        {%- if hasParameters %}
-            {%- for parameterName, parameter in channel.parameters() %}
-        parameters.put("{{parameterName}}", {{parameterName | camelCase}}{% if parameter.schema().type() !== 'string'%}.toString(){% endif %});
-            {%- endfor %}
-        {%- endif %}
-        return replaceParameters("{{channelName}}", parameters);
-    }
-    {%- endif %}
-{%- endfor %}
-    private String replaceParameters(String topic, Map<String, String> parameters) {
-        if (parameters != null) {
-            String compiledTopic = topic;
-            for (String key : parameters.keySet()) {
-                compiledTopic = compiledTopic.replace("{" + key + "}", parameters.get(key));
+        // Asynchronous send with callback
+        ListenableFuture<SendResult<Integer, Object>> future = kafkaTemplate.send(message);
+        future.addCallback(new ListenableFutureCallback<SendResult<Integer, Object>>() {
```

Comment:
ListenableFutureCallback is [deprecated](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/concurrent/ListenableFutureCallback.html)